File: dpdp-accelerator/components/org.wso2.dpdp.accelerator.event.notifications.dao/src/main/java/org/wso2/dpdp/accelerator/event/notifications/dao/model/Event.java
/**
 * Copyright (c) 2026, WSO2 LLC. (https://www.wso2.com).
 * <p>
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 *     http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package org.wso2.dpdp.accelerator.event.notifications.dao.model;

import java.sql.Timestamp;
import java.util.List;

public class Event {

    private String eventId;
    private String orgId;
    private String groupId;
    private String topicId;
    private String payload;
    private Timestamp createdAt;
    private List<String> purposes;

    private String topic;
    private int deliveriesCount;

    public Event() {
    }

    public Event(String eventId, String orgId, String groupId, String topicId, String payload, Timestamp createdAt) {
        this.eventId = eventId;
        this.orgId = orgId;
        this.groupId = groupId;
        this.topicId = topicId;
        this.payload = payload;
        this.createdAt = createdAt;
    }

    public String getTopic() {
        return topic;
    }

    public void setTopic(String topic) {
        this.topic = topic;
    }

    public int getDeliveriesCount() {
        return deliveriesCount;
    }

    public void setDeliveriesCount(int deliveriesCount) {
        this.deliveriesCount = deliveriesCount;
    }

    public String getEventId() {
        return eventId;
    }

    public void setEventId(String eventId) {
        this.eventId = eventId;
    }

    public String getOrgId() {
        return orgId;
    }

    public void setOrgId(String orgId) {
        this.orgId = orgId;
    }

    public String getGroupId() {
        return groupId;
    }

    public void setGroupId(String groupId) {
        this.groupId = groupId;
    }

    public String getTopicId() {
        return topicId;
    }

    public void setTopicId(String topicId) {
        this.topicId = topicId;
    }

    public String getPayload() {
        return payload;
    }

    public void setPayload(String payload) {
        this.payload = payload;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    public List<String> getPurposes() {
        return purposes;
    }

    public void setPurposes(List<String> purposes) {
        this.purposes = purposes;
    }
}

File: dpdp-accelerator/react-apps/consent-portal/frontend/src/types/event.ts
/*
 * Copyright (c) 2026, WSO2 LLC. (https://www.wso2.com).
 *
 * WSO2 LLC. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

export interface EventRecord {
  deliveryId?: string
  eventId: string
  subscriptionId?: string
  orgId?: string
  groupId?: string
  topic?: string
  topicId?: string
  payload?: string
  purposes?: string[]
  currentStatus?: string
  deliveryMode?: string
  deliveriesCount?: number
  occurredAt: string | number
  createdAt?: string | number
}

export interface EventDetailRecord extends EventRecord {
  deliveries?: EventRecord[]
}

export interface EventListQueryParams {
  limit: number
  offset: number
  search?: string
  status?: string
  topic?: string
  subscriptionId?: string
  groupId?: string
  purposes?: string
}

export interface EventListResponse {
  items: EventRecord[]
  total: number
}

export interface EventInput {
  topic: string
  groupId?: string
  purposes?: string[]
  payload: Record<string, unknown>
}

export interface EventFilters {
  search: string
  status: string
  topic: string
  groupId: string
  subscriptionId: string
}
